x-pack/plugins/endpoint/public/embeddables/resolver/store/camera/action.ts

```diff
@@ -24,11 +24,17 @@ interface UserClickedZoomIn {
 
 interface UserZoomed {
   readonly type: 'userZoomed';
-  /**
-   * A value to zoom in by. Should be a fraction of `1`. For a `'wheel'` event when `event.deltaMode` is `'pixel'`,
-   * pass `event.deltaY / -renderHeight` where `renderHeight` is the height of the Resolver element in pixels.
-   */
-  payload: number;
+  payload: {
+    /**
+     * A value to zoom in by. Should be a fraction of `1`. For a `'wheel'` event when `event.deltaMode` is `'pixel'`,
+     * pass `event.deltaY / -renderHeight` where `renderHeight` is the height of the Resolver element in pixels.
+     */
+    zoomChange: number;
+    /**
+     * Time when this action was dispatched.
+     */
+    time: Date;
```

Comment:
TODO justify why we don't use middleware to add timestamps

Reply:
Pros for metadata:
you don't have to type in `new Date()` when creating actions

Cons for metadata:
extra complexity.
you have to install middleware for the reducer to work, that includes tests
hidden complexity: not obvious that action's aren't pure (they aren't pure because their meaning is effected by the time, and conditions, under which they are dispatched.)